### PR TITLE
Fix logits processor bugs with spec dec

### PR DESCRIPTION
**Pull request by awni**
Fixes ~3 bugs with spec-dec:

- Pass 2D logits to logits processor
- Fix previous token book-keeping
- Incorrect normalization of log probs for base model

### CODE REVIEW

**Comment by awni** [comment]
CC @mattjcly

**Review by angeloskath — APPROVED** [review]
🙏
